test/integration/Resends.test.js

```diff
@@ -0,0 +1,170 @@
+        }, 50000)
+
+        it('resend last using subscribe function', async () => {
+            for (let i = 0; i < TEST_REPEATS; i++) {
+                const receivedMessages = []
+
+                // eslint-disable-next-line no-await-in-loop
+                const sub = client.subscribe(
+                    {
+                        stream: stream.id,
+                        resend: {
+                            last: MAX_MESSAGES,
+                        },
+                    },
+                    (message) => {
+                        receivedMessages.push(message)
+                    },
+                )
+
+                // eslint-disable-next-line no-loop-func
+                sub.once('resent', () => {
+                    expect(receivedMessages).toStrictEqual(publishedMessages)
+                })
+
+                // eslint-disable-next-line no-await-in-loop
+                await waitForCondition(() => receivedMessages.length === MAX_MESSAGES, 10000)
+            }
+        }, 50000)
+
+        it('resend last using subscribe and publish messages after resend', async () => {
+            const receivedMessages = []
+
+            client.subscribe({
+                stream: stream.id,
+                resend: {
+                    last: MAX_MESSAGES,
+                },
+            }, (message) => {
+                receivedMessages.push(message)
+            })
+
+            // wait for resend MAX_MESSAGES
+            await waitForCondition(() => receivedMessages.length === MAX_MESSAGES, 10000)
+            expect(receivedMessages).toStrictEqual(publishedMessages)
+
+            // publish after resend, realtime subscription messages
+            for (let i = MAX_MESSAGES; i < MAX_MESSAGES * 2; i++) {
+                const message = {
+                    msg: `message${i}`,
+                }
+
+                // eslint-disable-next-line no-await-in-loop
+                await client.publish(stream.id, message)
+                publishedMessages.push(message)
+            }
+
+            await waitForCondition(() => receivedMessages.length === MAX_MESSAGES * 2, 10000)
+            expect(receivedMessages).toStrictEqual(publishedMessages)
+        }, 30000)
+
+        it('resend last using subscribe and publish realtime messages', async () => {
+            const receivedMessages = []
+
+            const sub = client.subscribe({
+                stream: stream.id,
+                resend: {
+                    last: MAX_MESSAGES,
+                },
+            }, (message) => {
+                receivedMessages.push(message)
+            })
+
+            sub.on('subscribed', async () => {
+                for (let i = MAX_MESSAGES; i < MAX_MESSAGES * 2; i++) {
+                    const message = {
+                        msg: `message${i}`,
+                    }
+
+                    // eslint-disable-next-line no-await-in-loop
+                    await client.publish(stream.id, message)
```

Comment:
maybe `client.publish(stream.id, message).then(() => publishedMessages.push(message))` so it publishes in parallel and the half-minute timeout could be lowered?

Reply:
I tried, but I need order :) Maybe it's another issue...